ads/src/main/java/com/hms/lib/commonmobileservices/ads/splash/CommonSplashAdView.kt

```diff
@@ -0,0 +1,190 @@
+        get() {
+            val config = resources.configuration
+            return if (config.orientation == Configuration.ORIENTATION_LANDSCAPE) {
+                ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE
+            } else {
+                ActivityInfo.SCREEN_ORIENTATION_PORTRAIT
+            }
+        }
+
+    /**
+     * Handler to manage the timeout for the splash ad.
+     * If the ad is not loaded within the specified timeout, this handler will trigger the `jump` method to proceed.
+     */
+    private val timeoutHandler = Handler(Looper.getMainLooper()) {
+        if (it.what == MSG_AD_TIMEOUT) {
+            if (hasWindowFocus()) {
+                jump()
+            }
+        }
+        false
+    }
+
+    /**
+     * Initializes the CommonSplashAdView with attributes from XML.
+     * This block reads the custom attributes defined for the view and sets the relevant properties.
+     */
+    init {
+        val typedArray = context.obtainStyledAttributes(
+            attrs,
+            R.styleable.CommonSplashAdViewParams, 0, 0
+        )
+        gmsAdUnitId =
+            typedArray.getString(R.styleable.CommonSplashAdViewParams_gms_splash_ad_unit_id)
+        hmsAdUnitId =
+            typedArray.getString(R.styleable.CommonSplashAdViewParams_hms_splash_ad_unit_id)
+        screenOrientation =
+            typedArray.getInt(R.styleable.CommonSplashAdViewParams_screen_orientation, 0)
+        typedArray.recycle()
+    }
+
+    /**
+     * Loads the appropriate splash ad based on the mobile service type.
+     * @param callback The callback to be invoked when the ad is loaded or fails to load.
+     */
+    fun load(
+        callback: SplashAdLoadCallback,
+    ) {
+        when (Device.getMobileServiceType(context)) {
+            MobileServiceType.GMS -> {
+                AppOpenAd.load(
+                    context,
+                    gmsAdUnitId ?: "",
+                    AdRequest.Builder().build(),
+                    object : AppOpenAd.AppOpenAdLoadCallback() {
+                        override fun onAdLoaded(appOpenAd: AppOpenAd) {
+                            val googleAppOpenAdFactory =
+                                SplashAdFactory.createFactory<GoogleAppOpenAd, AppOpenAd>(
+                                    appOpenAd
+                                )
+                            googleAppOpenAdFactory.create().let(callback::onSplashAdLoaded)
+                        }
+
+                        override fun onAdFailedToLoad(loadAdError: LoadAdError) {
+                            callback.onAdLoadFailed(loadAdError.toString())
+                        }
+                    }
+                )
+            }
+
+            MobileServiceType.HMS -> {
+                hmsSplashView = SplashView(context)
+                val adRequestParams = AdParam.Builder().build()
+                hmsSplashView.load(
+                    hmsAdUnitId ?: "",
+                    screenOrientation,
```

Comment:
Is this value found both with the dynamic function and manually from attributes? you must choose one :)

Reply:
✔